sky/engine/core/dart/window.dart

```diff
@@ -18,41 +23,102 @@ class WindowPadding {
 class Locale {
-  Locale(this.languageCode, this.countryCode);
+  const Locale(this.languageCode, this.countryCode);
 
+  /// The primary language subtag for the locale.
   final String languageCode;
+
+  /// The region subtag for the locale.
   final String countryCode;
 
+  bool operator ==(dynamic other) {
+    if (identical(this, other))
+      return true;
+    if (other is! Locale)
+      return false;
+    final Locale typedOther = other;
+    return languageCode == typedOther.languageCode
+        && countryCode == typedOther.countryCode;
+  }
+
+  int get hashCode {
+    int result = 373;
+    result = 37 * result + languageCode.hashCode;
+    result = 37 * result + countryCode.hashCode;
+    return result;
+  }
+
   String toString() => '${languageCode}_$countryCode';
 }
 
+/// The most basic interface to the host operating system's user interface.
 class Window {
   Window._();
 
+  /// The number of device pixels for each logical pixel. This number might not
+  /// be a power of two. Indeed, it might not even be an integer. For example,
+  /// the Nexus 6 has a device pixel ratio of 3.5.
   double get devicePixelRatio => _devicePixelRatio;
   double _devicePixelRatio;
 
+  /// The dimensions of the rectangle into which the application will be drawn,
+  /// in logical pixels.
   Size get size => _size;
   Size _size;
 
+  /// The number of pixels on each side of the display rectangle into which the
+  /// application can render, but over which the operating system will likely
+  /// place system UI (such as the Android system notification area).
   WindowPadding get padding => _padding;
   WindowPadding _padding;
 
+  /// A callback that is invoked whenever the [devicePixelRatio], [size], or
+  /// [padding] values change.
+  VoidCallback onMetricsChanged;
+
+  /// The system-reported locale. This establishes the language and formatting
+  /// conventions that application should, if possible, use to render their user
+  /// interface.
   Locale get locale => _locale;
   Locale _locale;
 
+  /// A callback that is invoked whenever [locale] changes value.
+  VoidCallback onLocaleChanged;
+
+  /// A callback that is invoked to notify the application that it is an
+  /// appropriate time to provide a scene using the [SceneBuilder] API and the
+  /// [render()] method. When possible, this is driven by the hardware VSync
+  /// signal. This is only called if [scheduleFrame()] has been called since the
+  /// last time this callback was invoked.
   _FrameCallback onBeginFrame;
-  _EventCallback onEvent; // TODO(abarth): Remove.
+
+  /// A callback that is invoked when pointer data is available. The data is
+  /// provided in the form of a raw byte stream containing an encoded mojo
+  /// PointerPacket.
   _PointerPacketCallback onPointerPacket;
-  VoidCallback onMetricsChanged;
-  VoidCallback onLocaleChanged;
 
+  /// The route or path that the operating system requested when the application
+  /// was launched.
   String defaultRouteName;
```

Comment:
I don't know, I missed that it was. It has an odd name for a globally-mutable field.